### PR TITLE
Update(UT): Refactor init functions

### LINKED ISSUE #4192: Test existing firestore rules - Refactor Firestore init code to avoid open security rules
Refactor firestore init.

### PR DESCRIPTION
**Pull request by xmano**
Fix #4192 

`let db: any`

type for db inside tests still gives problem.

### CODE REVIEW

**Comment by xmano** [comment]
Getting permission error - everything seems ok with creation of movie @phpgeekfr  need your inputs here:

```
  ● Movies Rules Tests › With User in org › user valid org, with create permission for org should be able to create movie

    FirebaseError: 7 PERMISSION_DENIED:
    false for 'create' @ L21, Service call error. Function: [getAfter], Argument: [path_value {
      segments {
        simple: "databases"
      }
      segments {
        simple: "(default)"
      }
      segments {
        simple: "documents"
      }
      segments {
        simple: "permissions"
      }
      segments {
        simple: "O001"
      }
      segments {
        simple: "documentPermissions"
      }
      segments {
        simple: "MI-007"
      }
    }
    ]. for 'create' @ L275

      at new FirestoreError (node_modules/@firebase/rules-unit-testing/node_modules/@firebase/firestore/src/util/error.ts:216:5)
      at ClientDuplexStreamImpl.<anonymous> (node_modules/@firebase/rules-unit-testing/node_modules/@firebase/firestore/src/platform/node/grpc_connection.ts:232:15)
      at Object.onReceiveStatus (node_modules/@grpc/grpc-js/src/client.ts:670:18)
      at Object.onReceiveStatus (node_modules/@grpc/grpc-js/src/client-interceptors.ts:397:48)
      at Http2CallStream.outputStatus (node_modules/@grpc/grpc-js/src/call-stream.ts:230:22)
      at Http2CallStream.maybeOutputStatus (node_modules/@grpc/grpc-js/src/call-stream.ts:280:14)
      at Http2CallStream.endCall (node_modules/@grpc/grpc-js/src/call-stream.ts:263:12)
      at Http2CallStream.handleTrailers (node_modules/@grpc/grpc-js/src/call-stream.ts:412:10)

```

What is wrong here?

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 1]
```diff
@@ -30,21 +30,20 @@ function setRules(projectId: string, rulePath: string) {
   });
 }
 
-function setData(db: Firestore, dataDB: Record<string, Object>) {
+function setData(projectId: string, dataDB: Record<string, Object>) {
+  const app = initializeAdminApp({ projectId});
+  const dba = app.firestore();
```
Comment (GrandSchtroumpf) [inline apps/backend-ops/src/firestore-rules.spec.ts:35]: You can call it "db". The "a" (I guess for "admin") is quite curious as you're in a single scoped function.

File: apps/backend-ops/src/firestore-rules.spec.ts [hunk 2]
```diff
@@ -1,24 +1,24 @@
-﻿import { apps, assertFails, assertSucceeds, initializeTestApp, 
-        loadFirestoreRules, firestore } from '@firebase/rules-unit-testing';
+﻿import { apps, assertFails, assertSucceeds, initializeTestApp,
+        loadFirestoreRules, initializeAdminApp } from '@firebase/rules-unit-testing';
 import { testFixture } from './fixtures/data';
 import fs from 'fs';
+import { TokenOptions } from '@firebase/rules-unit-testing/dist/src/api';
 
-type Firestore = ReturnType<typeof initFirestoreApp>;
+//type Firestore = ReturnType<typeof initFirestoreApp> //<-- does not work!
```
Comment (GrandSchtroumpf) [inline apps/backend-ops/src/firestore-rules.spec.ts:7]: Why does it not work ?

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 1 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts:36]: changed to db

File: apps/backend-ops/src/firestore-rules.spec.ts (hunk 2 above)
Comment (xmano) [reply apps/backend-ops/src/firestore-rules.spec.ts]: Now changed to `type Firestore = ExtractPromise<PromiseFirestore>`

**Comment by xmano** [comment]
image

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**